### DISCUSSION TITLE
last 2 days, blank screen after image finish and no preview, nothing moved to folder

### DISCUSSION THREAD

**Discussion by lgwjames** [lllyasviel/Fooocus#2288] [Q&A]
here is the error

/Users/Quad-Core/Fooocus/modules/anisotropic.py:132: UserWarning: The operator 'aten::std_mean.correction' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1699448805816/work/aten/src/ATen/mps/MPSFallback.mm:13.)
  s, m = torch.std_mean(g, dim=(1, 2, 3), keepdim=True)
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [16:09<00:00, 32.32s/it]
Requested to load AutoencoderKL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 1.59 seconds
Traceback (most recent call last):
  File "/Users/Quad-Core/Fooocus/modules/async_worker.py", line 833, in worker
    handler(task)
  File "/Users/Quad-Core/miniconda3/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Quad-Core/miniconda3/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Quad-Core/Fooocus/modules/async_worker.py", line 812, in handler
    log(x, d)
  File "/Users/Quad-Core/Fooocus/modules/private_logger.py", line 78, in log
    existing_split = open(html_name, 'r', encoding='utf-8').read().split('')
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen codecs>", line 322, in decode
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x9f in position 1: invalid start byte
Total time: 1015.39 seconds

yes^ time slow on my amd mac,

**Comment by mashb1t** [comment]
@lgwjames seems like you've modified the private_logger.py file in line 78, content should not be split by empty string, but `'<!--fooocus-log-split-->'`

You can also delete the file so python directly writes to a new file.

**Comment by lgwjames** [comment]
so i have solved that, but now keep getting the modules/anisotropic.py:132: UserWarning: The operator 'aten::std_mean.correction' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1699448805816/work/aten/src/ATen/mps/MPSFallback.mm:13.)
s, m = torch.std_mean(g, dim=(1, 2, 3), keepdim=True)

was ok, done a fresh install as well and times are slow today

**Reply by mashb1t** [reply]
Was it working before and what has changed since 2 days?

DirectML on Mac is not officially supported. Please follow the Mac installation instructions, not AMD, if you haven't done already.
There is an open issue on the DirectML repo for a similar issue: https://github.com/microsoft/DirectML/issues/536

**Comment by lgwjames** [comment]
i was on 864 and now it seems 865, images took 12-18 mins

**Reply by mashb1t** [reply]
Can you please provide the full terminal log for further analysis?

There has been no change to any mean function calls, only to lcm calculation in that regard. Feel free to jump back and check out tag 2.1.864, use the launch.py instead of entry_with_update.py and check if this is reproducible.

**Reply by lgwjames** [reply]
> Last login: Wed Feb 21 17:31:52 on ttys000
> You have mail.
> /Users/Quad-Core/Desktop/fooocus.command ; exit;
> (base) Quad-Core@iMac ~ % /Users/Quad-Core/Desktop/fooocus.command ; exit;
> Already up-to-date
> Update succeeded.
> [System ARGV] ['entry_with_update.py', '--theme', 'dark', '--disable-offload-from-vram']
> Python 3.11.5 (main, Sep 11 2023, 08:19:27) [Clang 14.0.6 ]
> Fooocus version: 2.1.865
> Running on local URL:  http://127.0.0.1:7865
> 
> To create a public link, set `share=True` in `launch()`.
> Total VRAM 65536 MB, total RAM 65536 MB
> Set vram state to: SHARED
> Device: mps
> VAE dtype: torch.float32
> Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --attention-split
> Refiner unloaded.
> model_type EPS
> UNet ADM Dimension 2816
> Using split attention in VAE
> Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
> Using split attention in VAE
> extra {'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.text_projection'}
> Base model loaded: /Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors
> Request to load LoRAs [['sd_xl_offset_example-lora_1.0.safetensors', 0.1], ['PerfectEyesXL.safetensors', 1.0], ['Perfect Hands v2.safetensors', 1.0], ['None', 1.0], ['None', 1.0]] for model [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors].
> Loaded LoRA [/Users/Quad-Core/Fooocus/models/loras/sd_xl_offset_example-lora_1.0.safetensors] for UNet [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 788 keys at weight 0.1.
> Loaded LoRA [/Users/Quad-Core/Fooocus/models/loras/PerfectEyesXL.safetensors] for UNet [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 722 keys at weight 1.0.
> Loaded LoRA [/Users/Quad-Core/Fooocus/models/loras/PerfectEyesXL.safetensors] for CLIP [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 264 keys at weight 1.0.
> Loaded LoRA [/Users/Quad-Core/Fooocus/models/loras/Perfect Hands v2.safetensors] for UNet [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 722 keys at weight 1.0.
> Loaded LoRA [/Users/Quad-Core/Fooocus/models/loras/Perfect Hands v2.safetensors] for CLIP [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 264 keys at weight 1.0.
> Fooocus V2 Expansion: Vocab with 642 words.
> Fooocus Expansion engine loaded for cpu, use_fp16 = False.
> Requested to load SDXLClipModel
> Requested to load GPT2LMHeadModel
> Loading 2 new models
> [Fooocus Model Management] Moving model(s) has taken 3.83 seconds
> App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
> [Parameters] Adaptive CFG = 7
> [Parameters] Sharpness = 2
> [Parameters] ADM Scale = 1.5 : 0.8 : 0.3
> [Parameters] CFG = 4.0
> [Parameters] Seed = 36780147093922757
> [Parameters] Sampler = dpmpp_2m_sde_gpu - karras
> [Parameters] Steps = 30 - 15
> [Fooocus] Initializing ...
> [Fooocus] Loading models ...
> Refiner unloaded.
> Request to load LoRAs [['sd_xl_offset_example-lora_1.0.safetensors', 0.1], ['PerfectEyesXL.safetensors', 1.0], ['Perfect Hands v2.safetensors', 1.0], ['JessicaL.safetensors', 1.0], ['None', 1.0]] for model [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors].
> Loaded LoRA [/Users/Quad-Core/Fooocus/models/loras/sd_xl_offset_example-lora_1.0.safetensors] for UNet [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 788 keys at weight 0.1.
> Loaded LoRA [/Users/Quad-Core/Fooocus/models/loras/PerfectEyesXL.safetensors] for UNet [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 722 keys at weight 1.0.
> Loaded LoRA [/Users/Quad-Core/Fooocus/models/loras/PerfectEyesXL.safetensors] for CLIP [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 264 keys at weight 1.0.
> Loaded LoRA [/Users/Quad-Core/Fooocus/models/loras/Perfect Hands v2.safetensors] for UNet [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 722 keys at weight 1.0.
> Loaded LoRA [/Users/Quad-Core/Fooocus/models/loras/Perfect Hands v2.safetensors] for CLIP [/Users/Quad-Core/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors] with 264 keys at weight 1.0.
> Requested to load SDXLClipModel
> Loading 1 new model
> unload clone 1
> [Fooocus Model Management] Moving model(s) has taken 3.69 seconds
> [Fooocus] Processing prompts ...
> [Wildcards] processing: __jessicaL__ , jessicaL sitting in a bmw car in the passenger seat smiling
> [Wildcards] JessicaL, A English rose woman, blonde hair, (blue eyes), ((light freckles around nose and cheeks)). 32dd breasts, perfect eyes, perfect hands , jessicaL sitting in a bmw car in the passenger seat smiling
> [Wildcards] processing: __neg__
> [Wildcards] Watermark
> [Fooocus] Preparing Fooocus text #1 ...
> [Prompt Expansion] JessicaL, A English rose woman, blonde hair, (blue eyes), ((light freckles around nose and cheeks)). 32dd breasts, perfect eyes, perfect hands, jessicaL sitting in a bmw car in the passenger seat smiling, detailed, sharp focus, candid, elegant, highly intricate, stunning, sublime, enhanced, rich light, creative
> [Fooocus] Encoding positive #1 ...
> [Fooocus] Encoding negative #1 ...
> [Parameters] Denoising Strength = 1.0
> [Parameters] Initial Latent shape: Image Space (896, 1152)
> Preparation time: 27.69 seconds
> [Sampler] refiner_swap_method = joint
> [Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
> Requested to load SDXL
> Loading 1 new model
> [Fooocus Model Management] Moving model(s) has taken 7.82 seconds
> huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
> To disable this warning, you can either:
> 	- Avoid using `tokenizers` before the fork if possible
> 	- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
>   0%|                                                                                                                                                                             | 0/30 [00:00<?, ?it/s]/Users/Quad-Core/Fooocus/modules/anisotropic.py:132: UserWarning: The operator 'aten::std_mean.correction' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at /Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/mps/MPSFallback.mm:13.)
>   s, m = torch.std_mean(g, dim=(1, 2, 3), keepdim=True)


running on a imac, 64gb ram, rx580 pro 8gb ram and ssd for output files, still 18mins a image

**Reply by mashb1t** [reply]
@lgwjames as already mentioned, MacOS is currently not officially supported and in general MacOS with AMD GPUs is not supported yet.

**Comment by lgwjames** [comment]
sorry for the delay, busy at work, i did a fresh install, on 2.1.865 and will test now, as not have 64gb ram in mac and brought some more

**Comment by lgwjames** [comment]
just a update, no errors after not adding the wildcard files

**Reply by mashb1t** [reply]
did you change a specific wildcard file? Maybe you have some invalid syntax in there which breaks the loading.
